qiskit_dynamics/models/hamiltonian_model.py

```diff
@@ -110,25 +112,70 @@ def rotating_frame(self) -> RotatingFrame:
+        # convert static operator to new frame setup
+        static_op = self.get_static_operator(in_frame_basis=True)
+        if static_op is not None:
+            static_op = -1j * static_op
+
+        new_static_operator = self.transfer_static_operator_between_frames(
+            static_op,
+            new_frame=new_frame,
+            old_frame=self.rotating_frame,
+        )
+
+        if new_static_operator is not None:
+            new_static_operator = 1j * new_static_operator
 
-        self._rotating_frame = RotatingFrame(rotating_frame)
+        # convert operators to new frame set up
+        new_operators = self.transfer_operators_between_frames(
+            self.get_operators(in_frame_basis=True),
+            new_frame=new_frame,
+            old_frame=self.rotating_frame,
+        )
 
-        if self._rotating_frame.frame_diag is not None:
-            self._operators = self.rotating_frame.operator_into_frame_basis(self._operators)
-            self._drift = self.rotating_frame.operator_into_frame_basis(self._drift)
-            self._drift = self._drift - Array(np.diag(1j * self.rotating_frame.frame_diag))
+        self._rotating_frame = new_frame
 
-        # Reset internal operator collection
-        if self.evaluation_mode is not None:
-            self.evaluation_mode = self.evaluation_mode
+        self._operator_collection = self.construct_operator_collection(
+            self.evaluation_mode, new_static_operator, new_operators
+        )
 
     def evaluate(self, time: float, in_frame_basis: Optional[bool] = False) -> Array:
         return -1j * super().evaluate(time, in_frame_basis=in_frame_basis)
 
     def evaluate_rhs(self, time: float, y: Array, in_frame_basis: Optional[bool] = False) -> Array:
         return -1j * super().evaluate_rhs(time, y, in_frame_basis=in_frame_basis)
+
+
+def is_hermitian(
+    operators: Union[Array, csr_matrix, List[csr_matrix]], tol: Optional[float] = 1e-10
+) -> bool:
+    """Validate that operators are Hermitian.
+
+    Args:
+        operators: Either a 2d array representing a single operator, a 3d array
+                   representing a list of operators, a csr_matrix, or a list
+                   of csr_matrix.
+        tol: Tolerance for checking zeros.
+
+    Returns:
+        bool: Whether or not the operators are Hermitian to within tolerance.
+
+    Raises:
+        QiskitError: If an unexpeted type.
+    """
+    if isinstance(operators, (np.ndarray, Array)):
+        adj = None
+        if operators.ndim == 2:
+            adj = np.transpose(np.conjugate(operators))
+        elif operators.ndim == 3:
+            adj = np.transpose(np.conjugate(operators), (0, 2, 1))
+        return np.linalg.norm(adj - operators) < tol
+    elif issparse(operators):
+        return spnorm(operators - operators.conj().transpose()) < tol
+    elif isinstance(operators, list) and issparse(operators[0]):
+        return all(spnorm(op - op.conj().transpose()) < tol for op in operators)
```

Comment:
This could potentially be faster if `all` was swapped to `any` and the logic flipped so it can terminate as soon as one fails
```suggestion
        return not any(spnorm(op - op.conj().transpose()) >= tol for op in operators)
```